File: C#/EntityFramework_ORM/DAL/Entities/Order_Detail.cs
using DAL.Interfaces;
using System;
using System.Collections.Generic;

namespace DAL.Entities;

public partial class Order_Detail : IEntity
{
    public int OrderID { get; set; }

    public int ProductID { get; set; }

    public decimal UnitPrice { get; set; }

    public short Quantity { get; set; }

    public float Discount { get; set; }

    public virtual Order Order { get; set; } = null!;

    public virtual Product Product { get; set; } = null!;
}

File: Go/GORM_ORM/dal/entities/orderDetail.go
package entities

type OrderDetail struct {
	OrderID   int     `gorm:"primaryKey;column:OrderID"`
	ProductID int     `gorm:"primaryKey;column:ProductID"`
	UnitPrice float64 `gorm:"type:money;default:0;column:UnitPrice"`
	Quantity  int     `gorm:"default:1;column:Quantity"`
	Discount  float64 `gorm:"default:0;column:Discount"`
	Order     Order   `gorm:"foreignKey:OrderID"`
	Product   Product `gorm:"foreignKey:ProductID"`
}

func (OrderDetail) TableName() string {
	return "Order Details"
}
